packages/web-client/tests/unit/models/workflow/workflow-session-test.ts

```diff
@@ -574,4 +605,76 @@ module('Unit | WorkflowSession model', function (hooks) {
   //TODO tests for MerchantSafe type
   //TODO tests for PrepaidCardSafe type
   //TODO tests for TransactionReceipt type
+
+  test('it stores information about updated and created date when persisting data for the first time', async function (assert) {
+    let ID = 'workflow-id-1';
+    let workflowPersistence = new WorkflowPersistence();
+    workflowPersistence.persistData(ID, {
+      name: 'EXAMPLE',
+      state: {},
+    });
+    let subject = new WorkflowSession({
+      workflowPersistence,
+      workflowPersistenceId: ID,
+    });
+    let initialMeta = subject.getMeta();
+    assert.equal(
+      initialMeta?.createdAt,
+      null,
+      'There is no created date when session is instantiated'
+    );
+    assert.equal(
+      initialMeta?.updatedAt,
+      null,
+      'There is no updated date when session is instantiated'
+    );
+
+    // store something so that meta is also updated
+    subject.setValue('arbitrary-key', 'arbitrary-value');
+
+    let updatedMeta = subject.getMeta();
+    assert.equal(
+      updatedMeta.createdAt,
+      startDateString,
+      `Created date has the appropriate ISO date string: ${updatedMeta.createdAt}`
+    );
+    assert.equal(
+      updatedMeta.updatedAt,
+      startDateString,
+      `Updated date has the appropriate ISO date string: ${updatedMeta.updatedAt}`
+    );
+  });
+
+  test('it can update last updated date', async function (assert) {
+    let ID = 'workflow-id-1';
+    let workflowPersistence = new WorkflowPersistence();
+    let startDateString = new Date().toISOString();
+    workflowPersistence.persistData(ID, {
+      name: 'EXAMPLE',
+      state: {},
+    });
+    let subject = new WorkflowSession({
+      workflowPersistence,
+      workflowPersistenceId: ID,
+    });
+
+    subject.setValue('arbitrary-key', 'arbitrary-value');
+
+    await clock.tickAsync(60000);
+    let updatedDateString = new Date().toISOString();
+
+    subject.setValue('arbitrary-key-2', 'arbitrary-value-2');
+
+    let meta = subject.getMeta();
+    assert.equal(
+      meta.createdAt,
+      startDateString,
+      `Created date has the start ISO date string: ${startDateString}`
+    );
+    assert.equal(
+      meta.updatedAt,
+      updatedDateString,
+      `Updated date has an updated ISO date string: ${updatedDateString}`
+    );
+  });
```

Comment:
Should there be a test that writes arbitrary data to `meta`?

Reply:
There probably should be, will add, thanks